### PR TITLE
Switch from Travis to GHA

### PR DESCRIPTION
**Pull request by mfn**
## Summary
Get rid of travis and replace it with Github Actions

The recent troubles with travis (PRs not getting updated properly, preventing them from merging) led me to move away from it, since GHA is available for free for public repos.

Everything from travis was moved, with the following changes:
- "tests" and "integration tests" were split up
- Removed testing of `dev-master`
  There's no (easy?) equivalent of Travis `allow_failures` and no one really cared/bothered with those anyway
- We already had the code analysis done on GHA, those have been updated slightly to match the newer version (action version bumps and such)

### TODO
- [x] Turn off Travis-integration before merging
  Can't merge anyway, as it requires the successful run of it which won't work with travis anymore as the `.travis.yml` is moreved here
- [ ] Optionally: after merge, enforce certain/all jobs to have to be green before merging

## Type of change
- [x] Misc. change (internal, infrastructure, maintenance, etc.)

### Checklist
Nothing to be done

### Links
- Fixes https://github.com/rebing/graphql-laravel/issues/635

### CODE REVIEW

File: .github/workflows/analysis.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+name: Code Analysis
```
Comment (mfn) [inline .github/workflows/analysis.yml:1]: Files was renamed from `ci.ym`, no idea why git can't detected this properly

File: README.md [hunk 2]
```diff
@@ -1,8 +1,6 @@
 # Laravel GraphQL
 
 [![Latest Stable Version](https://poser.pugx.org/rebing/graphql-laravel/v/stable)](https://packagist.org/packages/rebing/graphql-laravel)
-[![codecov](https://codecov.io/gh/rebing/graphql-laravel/branch/master/graph/badge.svg)](https://codecov.io/gh/rebing/graphql-laravel)
```
Comment (mfn) [inline README.md:4]: drive-by change: realized that this isn't relevant anymore, as code coverage was removed > year ago

File: tests/integration-lumen.sh [hunk 3]
```diff
@@ -2,18 +2,18 @@
 
 # Inspired by https://github.com/nunomaduro/larastan/blob/669b489e10558bd45fafc2429068fd4a73843802/tests/laravel-test.sh
 #
-# Create a fresh Kumen installation, install our package in it and run some
+# Create a fresh Lumen installation, install our package in it and run some
```
Comment (mfn) [inline tests/integration-lumen.sh:5]: 😅

**Comment by mfn** [comment]
@rebing note: if you approve of this 🙏 , you must remove the travis integration 🛑 

Otherwise no PR can pass, as the Travis integration will never turn green 💥

**Comment by mfn** [comment]
@rebing thanks, I think however there's still something enforcing the success return from travis, see the PR.

I can't access https://github.com/rebing/graphql-laravel/settings/branches , but can you check that in case there are branch protection rules for `master`, Travis is removed from them?

thanks

**Comment by rebing** [comment]
@mfn I removed the requirement check, try now

**Comment by mfn** [comment]
@rebing it worked 🎉
